src/funjson.c

```diff
@@ -635,18 +641,15 @@ FUNCTION(fun_json)
     jstr = cJSON_PrintUnformatted(arr);
     cJSON_Delete(arr);
     latin1 = utf8_to_latin1(jstr, -1, &len, 0, "json.string");
-    for (c = latin1; *c; c += 1) {
-      if (!isprint(*c) && !isspace(*c)) {
-        *c = '?';
-      }
-    }
 
-    safe_strl(latin1, len, buff, bp);
+    partialbuf = json_unescape_latin1string(latin1,len);
+    safe_strl(partialbuf, strlen(partialbuf), buff, bp);
```

Comment:
Use safe_str(foo) instead of safe_strl(foo, strlen(foo)) - safe_strl is only really useful if you already know the length of the string